samples/routing/custom/sample.md

```diff
@@ -0,0 +1,155 @@
+
+
+### Sales
+
+The Sales project mimics the back-end system where orders are accepted. Apart from the standard NServiceBus configuration it enables the custom automatic routing:
+
+snippet:EnableAutomaticRouting
+
+NOTE: In order to use this custom routing all published types need to be specified.
+
+In real-world scenarios NServiceBus is scaled out by deploying multiple physical instances of a single logical endpoint to multiple machines (e.g. `Sales` in this sample). For simplicity in this sample the scale-out is simulated by having two separate endpoints `Sales` and `Sales2`.
+
+
+### Shipping and Billing
+
+Shipping and Billing mimic other back-end systems.
+
+
+### Shared project
+
+The shared project contains definitions for messages and the custom routing logic. 
+
+
+### Custom automatic routing
+
+The automatic routing is based on the idea of endpoints exchanging information about types of messages they handle and types of messages they publish. In this sample they do it via a table in SQL Server database (each endpoint instance owns one row) but other solutions can be used (e.g. Consul). Following code is used inside the custom automatic routing feature to wire up all its components:
+
+snippet:Feature
+
+It creates a publisher and a subscriber for the routing information along with the communication object they use to exchange the information between endpoint instances. It also registers an additional behavior that makes sure all published types are properly advertised. The automatic routing discovery protocol has no master and works in a peer-to-peer manner so both publisher and subscriber need to be active in each endpoint instance.
+
+<!--
+http://www.planttext.com/planttext
+@startuml
+
+package "Client" {
+    component [Routng\nSubscriber] as C_S
+    Component [Routing\nPublisher] as C_P 
+}
+
+package "Sales.1" {
+    component [Routng\nSubscriber] as S1_S
+    Component [Routing\nPublisher] as S1_P 
+}
+
+package "Sales.2" {
+    component [Routng\nSubscriber] as S2_S
+    Component [Routing\nPublisher] as S2_P 
+}
+
+database "SQL Server\n" {
+    [Routing information] as RI
+}
+
+C_P -down-> RI
+RI -up-> C_S
+
+S1_P -down-> RI
+RI -up-> S1_S
+
+S2_P -down-> RI
+RI -up-> S2_S
+
+@enduml
+-->
+
+![Automatic routing design](design.png)
+
+In order to define custom routing it's necessary to add logical addresses of endpoints to the routing table, specify all physical instances of the endpoints, and specify all event publishers, as show in the following snippet
+
+snippet:AddDynamic
+
+The routing cache consists of four parts. The first one is used in the [endpoint mapping layer](http://docs.particular.net/nservicebus/messaging/routing#unicast-routing-endpoint-mapping-layer) and maps message types to sets of endpoint which are known (because they advertised this fact) to have handlers for these message types. 
```

Comment:
> http://docs.particular.net/nservicebus/messaging/routing#unicast-routing-endpoint-mapping-layer

should be (nservicebus/messaging/routing#unicast-routing-endpoint-mapping-layer) or (nservicebus/messaging/routing.md#unicast-routing-endpoint-mapping-layer)